src/dune_file.ml

```diff
@@ -1065,12 +1065,18 @@ module Library = struct
 
   let is_virtual t = Option.is_some t.virtual_modules
 
-  let main_module_name t =
+  module Main_module_name = struct
+    type t =
+      | This of Module.Name.t option
+      | Inherited_from of (Loc.t * Lib_name.t)
```

Comment:
Looks good. I actually had a wip refactoring that introduced a type like:

```
module Module : sig
  module Name : sig
    module Main : sig
      type t = Named of Name.t | Inherited_from of (Loc.t * Lib_name.t)
    end
  ..
  end
  ..
end
```

and the return value for all these function was `Module.Name.Main.t option`. But I think that this basically equivalent. Maybe we should move this sub module to `module.ml` however.